commit 3d8959ab5c3934cbc49b07b62b7ff51b1c44698c
Author: Martin Gräßlin <mgraesslin@kde.org>
Date:   Mon Mar 25 15:18:58 2013 +0100

    Adding an activitiesChanged signal to Toplevel
    
    Only emitted from Client.
    
    REVIEW: 109706
    BUG: 317366
    FIXED-IN: 4.10.3

diff --git a/kwin/client.cpp b/kwin/client.cpp
index a694cdd..eab7c5a 100644
--- a/kwin/client.cpp
+++ b/kwin/client.cpp
@@ -1571,6 +1571,7 @@ void Client::updateActivities(bool includeTransients)
         m_blockedActivityUpdatesRequireTransients |= includeTransients;
         return;
     }
+    emit activitiesChanged(this);
     m_blockedActivityUpdatesRequireTransients = false; // reset
     if (includeTransients)
         workspace()->updateOnAllActivitiesOfTransients(this);
diff --git a/kwin/toplevel.h b/kwin/toplevel.h
index 3714908..b11907a 100644
--- a/kwin/toplevel.h
+++ b/kwin/toplevel.h
@@ -150,7 +150,7 @@ class Toplevel
      * See http://standards.freedesktop.org/wm-spec/wm-spec-latest.html .
      */
     Q_PROPERTY(int windowType READ windowType)
-    Q_PROPERTY(QStringList activities READ activities)
+    Q_PROPERTY(QStringList activities READ activities NOTIFY activitiesChanged)
     /**
      * Whether this Toplevel is managed by KWin (it has control over its placement and other
      * aspects, as opposed to override-redirect windows that are entirely handled by the application).
@@ -322,6 +322,7 @@ signals:
      * schedule a repaint of the scene.
      **/
     void needsRepaint();
+    void activitiesChanged(KWin::Toplevel* toplevel);
 
 protected:
     virtual ~Toplevel();
